tests/test_component.py

```diff
@@ -447,7 +447,7 @@ class MyComponent(component.Component):
             template_name = "simple_template.html"
 
             def get(self, request, *args, **kwargs):
-                return self.render_to_response({"variable": "test"})
+                return HttpResponse(self.render({"variable": "test"}))
```

Comment:
Why is this better?

Reply:
I wasn't sure about it. I ended up concluding that it wasn't worthwhile to add another method that was just a HttpResponse wrapper. What do you think?

Reply:
Hmm. I think I actually like it. It saves you from one import per component file, and it mimics the Django [shortcut render_to_response](https://docs.djangoproject.com/en/2.2/topics/http/shortcuts/#render-to-response). It's a small convenience I think.

Reply:
For security purposes, I was thinking about escaping the content that goes into the slots by default:

```python
    def render(
        self,
        context_data: Dict[str, Any],
        slots_data: Optional[Dict[SlotName, str]] = None,
        escape_slots_content: Optional[bool] = True,
    ) -> str:
        context = Context(context_data)
        template = self.get_template(context)

        if slots_data:
            self._fill_slots(slots_data, escape_slots_content)

        updated_filled_slots_context: FilledSlotsContext = (
            self._process_template_and_update_filled_slot_context(
                context, template
            )
        )
        with context.update(
            {FILLED_SLOTS_CONTENT_CONTEXT_KEY: updated_filled_slots_context}
        ):
            return template.render(context)

    def _fill_slots(
        self,
        slots_data: Dict[SlotName, str],
        escape_content: bool = True,
    ):
        """Fill component slots outside of template rendering."""
        self.fill_content = [
            (
                slot_name,
                TextNode(escape(content) if escape_content else content),
                None,
            )
            for (slot_name, content) in slots_data.items()
        ]
```

What do you think?

Reply:
Ugh, security. Yes, we really should think this through. My thoughts:

* Is rendering a component with render_to_response() protected already?
* **Not** escaping slots would be very dangerous, you are right. This definitely needs to be fixed before merging.
* Protential problem: wouldn't it be common to render a slot with HTML contents, the same contents that we wouldn't want a user to have access to? How should we handle this?
* A couple of tests that make sure we are not passing <script> tags unescaped by default would probably be good.

Reply:
> Is rendering a component with render_to_response() without slots protected already?

Yes, I added a test to verify that passing insecure parameters to the context is safe. From what I could gather, the context is processed with [this](https://github.com/django/django/blob/e412d85b4626bc56eb25206a40c3529162ce5dfc/django/template/context.py#L281) and that escapes the content by default. 

> Not escaping slots would be very dangerous, you are right. This definitely needs to be fixed before merging.
> Protential problem: wouldn't it be common to render a slot with HTML contents, the same contents that we wouldn't want a user to have access to? How should we handle this?

I made it the default to escape the content passed in slots but added a parameter so that users can control this. I also added a mention of this in the README, which hopefully is enough for users to understand the implications of this.

> A couple of tests that make sure we are not passing <script> tags unescaped by default would probably be good.

✅

Reply:
Excellent.